waku/v2/node/storage/message/waku_message_store.nim

```diff
@@ -115,3 +130,60 @@ method getAll*(db: WakuMessageStore, onData: message_store.DataProc): MessageSto
 proc close*(db: WakuMessageStore) = 
   ## Closes the database.
   db.database.close()
+
+
+proc migrate*(db: SqliteDatabase, path: string = MIGRATION_PATH, tragetVersion: int64 = USER_VERSION): MessageStoreResult[bool] = 
+  ## compares the user_version of the db with the targetVersion 
+  ## runs migration scripts if the user_version is outdated (does not support down migration)
+  ## path points to the directory holding the migrations scripts
+  ## once the db is updated, it sets the user_version to the tragetVersion
+  
+  # read database version
+  let userVersion = db.getUserVerion()
+  debug "current db user_version", userVersion=userVersion
+  if userVersion.value == tragetVersion:
+    # already up to date
+    info "database is up to date"
+    ok(true)
+  
+  else:
+    # TODO check for the down migrations i.e., userVersion.value > tragetVersion
+    # fetch migration scripts
+    let migrationScriptsRes = getScripts(path)
+    if migrationScriptsRes.isErr:
+      return err("failed to load migration scripts")
+    let migrationScripts = migrationScriptsRes.value
+
+    # filter scripts based on their versions
+    let scriptsRes = migrationScripts.filterScripts(userVersion.value, tragetVersion)
+    if scriptsRes.isErr:
+      return err("failed to filter migration scripts")
+    
+    let scripts = scriptsRes.value
+    debug "scripts to be run", scripts=scripts
+    
+    
+    proc handler(s: ptr sqlite3_stmt) = 
+      discard
+    
+    # run the scripts
+    for script in scripts:
+      debug "script", script=script
+      # a script may contain multiple queries
+      let queries = script.splitScript()
```

Comment:
Is splitting the sql scripts strictly necessary? SQLite should be able to execute multiple scripts in a single execution

Reply:
Thanks! I tried passing the whole script and it worked. I recall trying it before, but it did not get through for some reason (I think the script had some comments in it which weren't properly handled by SQlite).